src/golang/lib/workflow/validate.go

```diff
@@ -100,6 +100,11 @@ func ValidateSchedule(
 	// Note that we still need to check for cycles when the schedule of an existing
 	// Workflow is modified.
 
+	if workflowID == schedule.SourceID {
+		// Cannot have a self-dependency
+		return http.StatusBadRequest, errors.New("Cannot allow cycles for cascading workflows.")
+	}
```

Comment:
I feel this can be incorporated into `checkForCycle` below because it seems like a special case. Isn't our algorithm already capturing this? If not can we update it to capture this?

Reply:
Addressed